com.unity.render-pipelines.universal/Runtime/2D/Light2DBatch.cs

```diff
@@ -0,0 +1,130 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Unity.Collections;
+using Unity.Mathematics;
+using UnityEngine.Experimental.Rendering.Universal.LibTessDotNet;
+using UnityEngine.Rendering;
+using UnityEngine.U2D;
+
+namespace UnityEngine.Experimental.Rendering.Universal
+{
+
+    internal struct ShapeMeshBatch
+    {
+        internal int hashCode;
+        internal int meshCount;
+        internal int startHash;
+        internal int endHash;
+    }
+
+    internal static class Light2DBatch
+    {
+
+        static Dictionary<ShapeMeshBatch, Mesh> s_BatchMeshes = new Dictionary<ShapeMeshBatch,Mesh>();
+
+        static List<CombineInstance> s_ActiveBatchMeshInstances = new List<CombineInstance>();
+
+        static HashSet<ShapeMeshBatch> s_ActiveBatchHashes = new HashSet<ShapeMeshBatch>();
+
+        static List<Mesh> s_MeshPool = new List<Mesh>();
+
+        static ShapeMeshBatch s_ActiveShapeMeshBatch = new ShapeMeshBatch();
+
+        static Material s_ActiveMaterial = null;
+
+        static int s_Batches = 0;
+        
+        static int s_MeshCombined = 0;
+        internal static Material sActiveMaterial => s_ActiveMaterial;
+        
+        internal static int sBatchCount => s_Batches;
+        
+        internal static int sMeshCount => s_MeshCombined;
+
+        internal static void StartScope()
```

Comment:
consider using the `using()` pattern?

Reply:
Simplified and fixed in 298c27472e100a9d7d5ace7fdb4825ae0b6ea6c5